### ISSUE TITLE
Wacom Tablet support

### ISSUE THREAD

**Issue by Grunerd** [linearmouse/linearmouse#577]
### OS

macOS 13.5 (Build 22G74)

### LinearMouse

v0.9.3

### Is your feature request related to a problem?

I own a CTH-480 (an older Wacom Tablet, but given how crappy their drivers are, it probably doesn't make a difference). And I use it in conjunction with a Logitech MX3. 

For that Logitech-Mouse, I have Reverse Scrolling activated. This however messes up panning with the Wacom-Tablet, as vertical up/down is reversed for that as well.

LinearMouse doesn't differentiate between the both as the Tablet is not being handled as a usual mouse, but through proprietary drivers. 

### Describe the solution you'd like

I wonder if there is a way to check how the mouse cursor is being moved and allow us to set up a vertical scrolling behaviour for "unknown third party mouse / tablet"-devices to enforce the wanted behaviour here. 

So basically the introduction of 3 states:
- Internal Trackpad Movement
- External Mouse Movement
- "Third Party" / "Software Movement"

And handling that third case as a separately discovered device. 

### Describe alternatives you've considered

So far, the only solution is to manually change the reverse scroll option whenever I work with the pen rather than my mouse. 

I've also looked up some scripts and apps that can change reverse scrolling via hotkey, but none of the solutions are "instant" or satisfying. 

### Additional context

Wacom's drivers are terrible and they are not going to change it.

**Comment by Grunerd** [comment]
<img width="962" alt="Screenshot 2023-08-18 at 08 35 26" src="https://github.com/linearmouse/linearmouse/assets/11262143/498a144b-7aad-47e5-b6a8-394c67ac30b1">

I just checked and apparently, the Wacom Tablet is recognized as "Intuos PTS" but never recognized as "active" even when in use. Manually switching to "Intuos PTS" and toggling the Reverse Scroll Option on and off doesn't change anything for the panning-behaviour, the MX3 remains the "active device". 

[LinearMouse.log](https://github.com/linearmouse/linearmouse/files/12376198/LinearMouse.log)

**Comment by lujjjh** [comment]
Thanks for your feedback.

I'm trying to resolve the issue of the tablet not being recognized as "active." Since I don't have access to such devices for testing, could you kindly check if the fix is effective?

You can download the development build from [here](https://github.com/linearmouse/linearmouse/suites/15257409962/artifacts/871733941). Please note that you will have to manually download [the most recent version](https://dl.linearmouse.org/latest/LinearMouse.dmg) to switch back to the release version so that you can receive future updates.

If this problem continues to persist or if any new issues arise, please activate the "Enable verbose logging" option. Afterward, perform a few actions on your Wacom Tablet and export the logs for further analysis.

**Comment by sealvessel** [comment]
Also, if it were to work with a wacom tablet(scrolling) it would need to support open table driver simultaneously.

**Comment by Grunerd** [comment]
> Thanks for your feedback.
> 
> I'm trying to resolve the issue of the tablet not being recognized as "active." Since I don't have access to such devices for testing, could you kindly check if the fix is effective?
> 
> You can download the development build from [here](https://github.com/linearmouse/linearmouse/suites/15257409962/artifacts/871733941). Please note that you will have to manually download [the most recent version](https://dl.linearmouse.org/latest/LinearMouse.dmg) to switch back to the release version so that you can receive future updates.
> 
> If this problem continues to persist or if any new issues arise, please activate the "Enable verbose logging" option. Afterward, perform a few actions on your Wacom Tablet and export the logs for further analysis.

Sorry that months have gone by without a response. A lot of stuff has happened on my end, but now I finally have the time to test. Sadly, the development build is not available anymore, but I can say that the issue persists with the latest build of LinearMouse that I have (0.9.6-beta.1)

Manually switching to the Intuos PTS doesn't change anything, all input is treated as it was still the MX Master 3S.

**Comment by lujjjh** [comment]
@Grunerd Thank you for responding!

I didn't merge the *fix* because I wasn't sure if it was effective. I've initiated [a rebuild](https://github.com/linearmouse/linearmouse/suites/16986739883/artifacts/970397933) and it would be great if you could test that build whenever you have a chance.

**Comment by Grunerd** [comment]
Thanks a lot for taking the time to initiate a new build. Sadly, it still doesn't recognise the change from any device to the Intuos.

Even in the verbose logging, inputs made via the intuos are logged as either Master MX 3S or the Apple Trackpad, depending on which device I used before. So half of the dragging movement and scrolling you see in the logs is the Wacom tablet falsely identified as either the one or the other.

I still assume that's because the Wacom tablets are not going through "HID"-compatible drivers. They seem to simply reposition the cursor via their "WacomTabletDriver", "TabletDriver" and "TouchDriver"-processes which run in the background.

[LinearMouse.log](https://github.com/linearmouse/linearmouse/files/12840047/LinearMouse.log)

**Comment by lujjjh** [comment]
> They seem to simply reposition the cursor via their "WacomTabletDriver", "TabletDriver" and "TouchDriver"-processes which run in the background.

Have you tried toggling the "Bypass events from other applications" option on in General settings?

**Comment by Grunerd** [comment]
> Have you tried toggling the "Bypass events from other applications" option on in General settings?

Sadly that makes no difference. Tried switching it on and off with all combinations of reboots and starting LinearMouse prior to plugging the tablet in or vice versa.

**Comment by smvueno** [comment]
Hi! I actually got the same problem with PTH-451.
When I disable "Auto switch to the active device" I can see the "Intuos5 touch s interface", but this is sadly not for the pen.

I manually switch the reverse scrolling off when using the tablet. But I sadly use mouse and pen all the time while switching between some apps and file management.

**Comment by luckman212** [comment]
Any Wacom owners ever find a solution? I have an MS-IntelliEye mouse with a wheel and I need to reverse it, but I want my Wacom to behave properly too. Can't seem to get these to cooperate!
